app/models/rule.rb

```diff
@@ -118,4 +118,9 @@ def latest_result(host, profile)
 
     test_result.rule_results.find_by(rule_id: id)
   end
+
+  def rule_group_ancestor_ids
+    # Rule should only have one rule_group parent
+    rule_groups.first.path_ids
+  end
```

Comment:
@skateman I think `rule_groups.first` should be good because it's impossible for a rule to have more than one rule_group parent which is what the `rule.rule_groups` association is

Reply:
I'm not entirely sure about that, can't a rule belong to multiple rule groups across different profiles? You should verify in the XCCDF specification and if this is true, you can rewrite the `has_many :rule_groups` to `has_one :rule_groups` which would render the array into a single element, thus not needing the `#first`.

Reply:
The hierarchical structure of rules and rule groups is the same for the entire benchmark so a `rule` will always have the same parent `rule_group` even if it belongs to different profiles under the benchmark. This is because the `Rule`/`Group` structure is determined by how the `<xccdf-1.2:Rule/>` and `<xccdf-1.2:Group>` tags are nested under each other in the datastream files which is just defined once for the entire benchmark. The rules are just "selected" by a Profile with the line `<xccdf-1.2:select idref="xccdf_org.ssgproject.content_rule_selinux_policytype" selected="true"/>` and groups are unselected with this line `<xccdf-1.2:select idref="xccdf_org.ssgproject.content_group_bootloader-zipl" selected="false"/>`.

But yeah, that's a good idea. I'll update it from `has_many` to `has_one` and that'll work better

Reply:
This might not work and would need simplification in the model as the RuleGroupRule table is obsolete. Try it and if it doesn't work, keep the `first`. We should surely do a remodel later, please create a ticket for it into the epic.

Reply:
Ah right, that's true. I just updated these 2 associations on Rule from `has_many` to `has_one` and it works correctly: 
```
has_one :rule_group_rule, dependent: :delete
has_one :rule_group, through: :rule_group_rule
```
But I can still create a ticket to do a remodel later if this is okay with you

Reply:
Yes

Reply:
Just created a ticket for that and added it to the epic